src/repositories/request-repository.ts

```diff
@@ -202,165 +249,88 @@ export class RequestRepository extends Repository<Request> {
         if (streamsToAnchor.length === 0) {
           logger.debug(`Not updating any requests to READY because there are no streams to anchor`)
           return []
         }
 
         // Anchor if we have enough streams or the earliest stream request is expired
-        const earliestIsExpired = streamsToAnchor[0].createdAt < anchoringDeadline
-        if (streamsToAnchor.length >= minStreamLimit || earliestIsExpired) {
+        const enoughStreams = streamsToAnchor.length >= minStreamLimit
+        const earliestIsExpired = streamsToAnchor[0].minCreatedAt < anchoringDeadline
+
+        if (enoughStreams || earliestIsExpired) {
           const streamIds = streamsToAnchor.map(({ streamId }) => streamId)
 
-          // using conditions in allRequestsToAnchorQuery, retrieves all requests associated with the retrieved streams
-          const requests = await allRequestsToAnchorQuery
-            .orderBy('request.createdAt', 'ASC')
-            .andWhere('request.streamId IN (:...streamIds)', { streamIds })
-            .getMany()
+          const requests = await findRequestsToAnchorForStreams(trx, streamIds, now)
 
-          const results = await this.updateRequests(
+          const updatedCount = await this.updateRequests(
             { status: RequestStatus.READY },
             requests,
-            transactionalEntityManager
+            {
+              connection: trx,
+            }
           )
 
           // if not all requests are updated
-          if (!results.affected || results.affected != requests.length) {
+          if (updatedCount != requests.length) {
             throw Error(
-              `A problem occured when updated requests to READY. Only ${results.affected}/${requests.length} requests were updated`
+              `A problem occured when updated requests to READY. Only ${updatedCount}/${requests.length} requests were updated`
             )
           }
 
           countRetryMetrics(requests, anchoringDeadline)
 
-          logger.debug(`Updated ${results.affected} requests to READY`)
+          logger.debug(`Updated ${updatedCount} requests to READY`)
 
           return requests
         }
 
         logger.debug(
-          `Not updating any requests to READY because there are not enough streams for a batch ${streamsToAnchor.length}/${minStreamLimit} and the earliest request is not expired (created at ${streamsToAnchor[0].createdAt})`
+          `Not updating any requests to READY because there are not enough streams for a batch ${streamsToAnchor.length}/${minStreamLimit} and the earliest request is not expired (created at ${streamsToAnchor[0].minCreatedAt})`
         )
 
         return []
-      })
-      .catch(async (err) => {
-        if (err?.code === REPEATED_READ_SERIALIZATION_ERROR) {
-          Metrics.count(METRIC_NAMES.DB_SERIALIZATION_ERROR, 1)
-          await Utils.delay(100)
-          return this.findAndMarkReady(maxStreamLimit, minStreamLimit)
```

Comment:
what happened to this retry logic?  Is it not needed anymore?

Reply:
good catch! i'll add that back in